docs/tutorials/todo-app/2-interacting-with-the-list.rst

```diff
@@ -0,0 +1,114 @@
+Making the list interactive
+============================
+
+So far, our TODO list is not very useful, since it's static. You can't update items,
+nor add or remove them.
+
+Receiving incoming data
+-----------------------
+
+Let's start by implementing a route handler that handles the creation of new items.
+In the previous step you have used the ``get`` decorator, which responds to the ``GET``
+HTTP method. In this case we want to react to ``POST`` requests, so we are going to use
+the corresponding ``post`` decorator.
+
+.. literalinclude:: examples/create/dict.py
+    :language: python
+    :linenos:
+
+Request data can be received via the ``data`` keyword. Litestar will recognise this, and
+supply the data being sent with the request via this parameter. As with the query
+parameters in the previous chapter, we use the type annotations to configure what type
+of data we expect to receive, and set up validation. In this case, Litestar will expect
+request data in the form of JSON and use the type annotation we gave it to convert it
+into the correct format.
+
+
+Using the interactive documentation to test a route
+++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Since our example now uses the ``POST`` HTTP method, you can no longer simply visit the
+URL in our browser and get a response. Instead, you can use the interactive
+documentation to send a ``POST`` request. Because of the OpenAPI schema generated by
+Litestar, Swagger will know exactly what kind of data to send. In this example, it will
+send a simple JSON object.
+
+.. figure:: images/swagger-post-dict-response.png
+
+    Sending a sample request to our ``add_item`` route reveals a successful response
+
+
+Improving the example with dataclasses
+++++++++++++++++++++++++++++++++++++++
+
+As in the previous chapter, this too can be improved by using
+:doc:`dataclasses <python:library/dataclasses>` instead of plain dicts.
+
+.. literalinclude:: examples/create/dataclass.py
+    :language: python
+    :linenos:
+
+
+This is not only easier on the eyes and adds more structure to the code, but also gives
+better interactive documentation; It will now present us with the field names and
+default values for the dataclass we have defined:
+
+.. figure:: images/swagger-dict-vs-dataclass.png
+
+    Documentation for the ``create_item`` route with ``data`` typed as a ``dict`` vs
+    ``dataclass``
+
+Using a dataclass also gives you better validation: Omitting a key such as ``title``
+will result in a useful error response:
+
+
+.. figure:: images/swagger-dataclass-bad-body.png
+
+    Sending a request without a ``title`` key fails
+
+
+Path parameters
+---------------
+
+The next task on the list is updating an item's status. For this, a way to refer to a
+specific item on the list is needed. This could be done using query parameters, but
+there's an easier, and more semantically coherent way of expressing this: Path
+parameters.
+
+.. code-block:: python
+
+    @get("/{name:str}")
```

Comment:
What happens when I have a route, `/test` and I add this dynamic route `/{blah: str}`, and visit `/` with `blah` set to `test`?

Reply:
The static route takes precedence. Do you think that's something we should add here? I intended to include more complex routing scenarios in a later chapter.

Reply:
It's a good question - but I'd say most beginners wouldn't be asking it at this point, better added elsewhere IMO.